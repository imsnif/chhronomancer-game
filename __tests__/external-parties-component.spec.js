import React from 'react'
import 'react-native'
import serverMock from '../server-mock'
import renderer from 'react-test-renderer'

jest.mock('mobx-react/native', () => require('mobx-react/custom'))

beforeEach(() => {
  jest.useFakeTimers()
  serverMock()
})

afterEach(() => {
  const clockStore = require('../stores/clock').default
  jest.useRealTimers()
  clockStore.destroy()
  jest.useFakeTimers()
})

test('ExternalParties component => renders correctly', () => {
  const ExternalParties = require('../components/external-parties').default
  const tree = renderer.create(
    <ExternalParties
      name='foo'
      total={2}
      image={{uri: 'http://foo.bar/baz'}}
      parties={[{id: 1, score: 2}]}
      onPress={() => {}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
