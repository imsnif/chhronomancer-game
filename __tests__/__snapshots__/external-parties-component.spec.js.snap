// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExternalParties component => renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#329932",
      "borderColor": "#329932",
      "borderRadius": 0,
      "borderWidth": 0.5,
      "flex": 1,
      "flexDirection": "column",
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "height": 35,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://foo.bar/baz",
        }
      }
      style={
        Object {
          "backgroundColor": "#141313",
          "borderColor": "black",
          "borderWidth": 1,
          "flex": 0,
          "height": 28,
          "margin": 1,
          "resizeMode": "stretch",
          "width": 26,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "green",
            "height": "100%",
            "opacity": 0.5,
            "width": "100%",
          }
        }
      />
    </Image>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 7,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "paddingLeft": 5,
          "textAlign": "left",
        }
      }
    >
      foo
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      numberOfLines={1}
      style={
        Object {
          "color": "#0e0e0e",
          "flex": 3,
          "fontFamily": "telegrama_raw",
          "fontSize": 30.607554623001164,
          "textAlign": "center",
        }
      }
    >
      2
    </Text>
  </View>
  <View
    style={
      Object {
        "backgroundColor": "#141313",
        "borderColor": "#141313",
        "borderRadius": 0,
        "borderWidth": 0.5,
        "flex": 13,
        "flexDirection": "column",
        "justifyContent": "space-between",
        "margin": 1,
      }
    }
  >
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "column",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
            "margin": 1,
            "maxHeight": 30,
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "http://10.0.0.6:3000/placeholders/nopic.png",
            }
          }
          style={
            Object {
              "flex": 0,
              "height": 26,
              "margin": 1,
              "resizeMode": "stretch",
              "width": 26,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "green",
                "height": "100%",
                "opacity": 0.5,
                "width": "100%",
              }
            }
          />
        </Image>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 3,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Object {
                "alignSelf": "center",
                "color": "green",
                "flex": 1,
                "fontFamily": "telegrama_raw",
                "paddingLeft": 3,
                "paddingRight": 5,
                "paddingTop": 3,
              }
            }
          >
            Aram
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignSelf": "center",
                "color": "green",
                "flex": 1,
                "fontFamily": "telegrama_raw",
                "paddingRight": 1,
                "paddingTop": 3,
              }
            }
          >
            2
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "padding": 5,
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "alignItems": "center",
                "backgroundColor": "#141313",
                "borderColor": "#329932",
                "borderRadius": 0,
                "borderWidth": 1,
                "flex": 1,
                "flexDirection": "row",
                "height": "100%",
                "justifyContent": "center",
                "width": "100%",
              },
            ]
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "color": "#329932",
                "flex": 1,
                "fontFamily": "telegrama_raw",
                "fontSize": 80,
                "textAlign": "center",
              }
            }
          >
            +
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;
